disk/src/com/treode/disk/edit/PickledRecord.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2014 Treode, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.treode.disk.edit
+
+import com.treode.async.Callback
+import com.treode.buffer.Output
+import com.treode.disk.RecordDescriptor
+
+private abstract class PickledRecord (
+    val cb: Callback [Unit]
+) {
+  def byteSize: Int
```

Comment:
We also need the type exposed here (spelled typ since "type" is a keyword in Scala).